c10/core/TensorImpl.h

```diff
@@ -2932,6 +2961,27 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   DispatchKeySet key_set_;
 
  private:
+  // Invariant: this is always null if a copy on write was never
+  // requested.
+  //
+  // This *may* be null if a copy on write was requested and this
+  // tensor is part of the original view family. Subsequent view
+  // families will have this set, but the original one only gets its
+  // value from the storage.
+  //
+  // This is asymmetrical, but it allows us to avoid the allocation
+  // and any refcount bumps until we actually need them.
+  intrusive_ptr<impl::cow::ShadowStorage> shadow_storage_;
```

Comment:
It would be good to have a direct pointer to the relevant README, so that someone who shows up here looking to axe out fields can figure out where to go

Reply:
added